pkg/cmd/docs-issue-generation/docs_issue_generation.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+type commit struct {
+	sha         string
+	title       string
+	releaseNote string
+}
+
+type pr struct {
+	number      int
+	mergeBranch string
+	commits     []commit
+}
+
+type ghSearch struct {
+	Items []struct {
+		Number      int `json:"number"`
+		PullRequest struct {
+			MergedAt *time.Time `json:"merged_at"`
+		} `json:"pull_request"`
+	} `json:"items"`
+}
+
+type ghPull struct {
+	Base struct {
+		Ref string `json:"ref"`
+	} `json:"base"`
+}
+
+type ghPullCommit struct {
```

Comment:
Some comments on these structs to capture what they are meant to represent would be good.

Reply:
Same with the functions, actually. For the best to include some comments if the name and signature doesn't tell the reader everything they need to know IMO.

Reply:
Added some comments explaining what the different blocks do.